presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -881,13 +884,22 @@ private String getQueryPlan(Explain node, ExplainType.Type planType, ExplainForm
     {
         switch (planFormat) {
             case GRAPHVIZ:
-                return queryExplainer.get().getGraphvizPlan(session, node.getStatement(), planType);
+                return queryExplainer.get().getGraphvizPlan(session, getStatementToExplain(node), planType);
             case TEXT:
-                return queryExplainer.get().getPlan(session, node.getStatement(), planType);
+                return queryExplainer.get().getPlan(session, getStatementToExplain(node), planType);
         }
         throw new IllegalArgumentException("Invalid Explain Format: " + planFormat.toString());
     }
 
+    private Statement getStatementToExplain(Explain node)
```

Comment:
I find the name here a little awkward. 'get' and 'to' have conflicting meanings in my mind. I propose the following alternatives: `unwrapStatementToExplainIfNeeded`, or `convertStatementToExplainIfNeeded` or you can drop the `IfNeeded` part.